subsys/bluetooth/host/smp.c

```diff
@@ -2721,6 +2721,10 @@ static uint8_t remote_sec_level_reachable(struct bt_smp *smp)
 {
 	struct bt_conn *conn = smp->chan.chan.conn;
 
+	if (get_encryption_key_size(smp) < BT_SMP_MAX_ENC_KEY_SIZE &&
+		IS_ENABLED(CONFIG_BT_SMP_SC_ONLY)) {
+		return BT_SMP_ERR_ENC_KEY_SIZE;
+	}
 	switch (conn->required_sec_level) {
```

Comment:
I suggested something else, the reason for my suggestion was to do all the checks required for SC_ONLY mode.
Now you have only the key size check, which mean that you wouldn't reject the pairing request if SC wasn't set, or it was trying to do an unauthenticated pairing.
With my suggestion we re-use the checks below for all of the 3 requirements.

The pairing would eventually have failed, rejecting it early here is an optimization.

```suggestion
	bt_security_t sec;
	
	if IS_ENABLED(CONFIG_BT_SMP_SC_ONLY) {
		sec = BT_SECURITY_L4;
	else {
		sec = conn->required_sec_level;
	}

	switch (sec) {
```